src/Core/Python/PythonDatatypeConverter.cc

```diff
@@ -451,50 +476,58 @@ namespace
   }
 }
 
-Variable SCIRun::Core::Python::convertPythonObjectToVariable(const boost::python::object& object)
+const std::string SCIRun::Core::Python::getClassName(const py::object& object)
+{
+  py::extract<py::object> extractor(object);
+  return py::extract<std::string>(extractor().attr("__class__").attr("__name__"));
+}
+
+Variable SCIRun::Core::Python::convertPythonObjectToVariable(const py::object& object)
 {
   /// @todo: yucky
+
+  auto classname = getClassName(object);
+  if (classname == "int")
   {
-    boost::python::extract<int> e(object);
-    if (e.check())
-    {
-      return makeVariable("int", e());
-    }
+    py::extract<int> e(object);
+    return makeVariable(classname, e());
   }
+  else if (classname == "float")
   {
-    boost::python::extract<double> e(object);
-    if (e.check())
-    {
-      return makeVariable("double", e());
-    }
+    py::extract<float> e(object);
+    return makeVariable(classname, e());
   }
+  else if (classname == "double")
   {
-    boost::python::extract<std::string> e(object);
-    if (e.check())
-    {
-      return makeVariable("string", e());
-    }
+    py::extract<double> e(object);
+    return makeVariable(classname, e());
   }
+  else if (classname == "str")
   {
-    boost::python::extract<bool> e(object);
-    if (e.check())
-    {
-      return makeVariable("bool", e());
-    }
+    py::extract<std::string> e(object);
+    return makeVariable(classname, e());
+  }
+  else if (classname == "bool")
+  {
+    py::extract<bool> e(object);
+    return makeVariable(classname, e());
   }
+  else if (classname == "list")
+  {
+    Variable::List vars(len(object));
+    for (int i = 0; i < len(object); ++i)
+      vars[i] = convertPythonObjectToVariable(object[i]);
+    return makeVariable(classname, vars);
+  }
+
   {
     DenseMatrixExtractor e(object);
-    if (e.check())
-    {
-      return makeDatatypeVariable(e);
-    }
+    if (e.check()) return makeDatatypeVariable(e);
   }
+
   {
     SparseRowMatrixExtractor e(object);
-    if (e.check())
-    {
-      return makeDatatypeVariable(e);
-    }
+    if (e.check()) return makeDatatypeVariable(e);
   }
```

Comment:
I was able to check the type without attempting an extraction on every type by checking for the python object's class name. Do the types DenseMatrix, SparseRowMatrix, and Field have expected class names of python objects I could use to clean up the remainder of this function?